Server/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```diff
@@ -70,374 +70,144 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Blip", "Element");
 }
 
-int CLuaBlipDefs::CreateBlip(lua_State* luaVM)
+std::variant<CBlip*, bool> CLuaBlipDefs::CreateBlip(lua_State* luaVM, CVector vecPosition, std::optional<std::uint8_t> icon, std::optional<std::uint8_t> size, std::optional<std::uint8_t> r, std::optional<std::uint8_t> g, std::optional<std::uint8_t> b, std::optional<std::uint8_t> a, std::optional<std::int32_t> ordering, std::optional<std::uint32_t> visibleDistance, std::optional<CElement*> visibleTo)
 {
-    CVector       vecPosition;
-    unsigned char ucIcon;
-    int           iSize;
-    SColorRGBA    color(255, 0, 0, 255);
-    int           iOrdering;
-    int           iVisibleDistance;
-    CElement*     pVisibleTo;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadVector3D(vecPosition);
-    argStream.ReadNumber(ucIcon, 0);
-    argStream.ReadNumber(iSize, 2);
-    argStream.ReadNumber(color.R, color.R);
-    argStream.ReadNumber(color.G, color.G);
-    argStream.ReadNumber(color.B, color.B);
-    argStream.ReadNumber(color.A, color.A);
-    argStream.ReadNumber(iOrdering, 0);
-    argStream.ReadNumber(iVisibleDistance, 16383);
-    if (argStream.NextIsBool() || argStream.NextIsNil())
-        pVisibleTo = NULL;
-    else
-        argStream.ReadUserData(pVisibleTo, m_pRootElement);
-
-    if (!CBlipManager::IsValidIcon(ucIcon))
-    {
-        argStream.SetCustomError("Invalid icon");
-    }
+    if (icon.has_value() && !CBlipManager::IsValidIcon(icon.value()))
+        throw std::invalid_argument("Invalid icon");
 
-    if (!argStream.HasErrors())
+    if (size.has_value() && (size.value() < 0 || size.value() > 25))
     {
-        CLuaMain* pLuaMain = g_pGame->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                unsigned char  ucSize = Clamp(0, iSize, 25);
-                short          sOrdering = Clamp(-32768, iOrdering, 32767);
-                unsigned short usVisibleDistance = Clamp(0, iVisibleDistance, 65535);
-
-                // Create the blip
-                CBlip* pBlip = CStaticFunctionDefinitions::CreateBlip(pResource, vecPosition, ucIcon, ucSize, color, sOrdering, usVisibleDistance, pVisibleTo);
-                if (pBlip)
-                {
-                    CElementGroup* pGroup = pResource->GetElementGroup();
-                    if (pGroup)
-                    {
-                        pGroup->Add(pBlip);
-                    }
-                    lua_pushelement(luaVM, pBlip);
-                    return 1;
-                }
-            }
-        }
+        m_pScriptDebugging->LogWarning(luaVM, SString("Blip size beyond 25 is no longer supported (got %i). It will be clamped between 0 and 25.", size.value()));
+        size = Clamp(0ui8, size.value(), 25ui8);
```

Comment:
`Clamp(0, size.value(), 25)`

Reply:
Clamp expects arguments of the same type. If I just give 0 and 25 it receives it as an int and there are compilation errors

Reply:
then cast 0 and 25 to uint8_t. `ui` suffix doesnt exist in C++ standard.

Reply:
> then cast 0 and 25 to uint8_t. `ui` suffix doesnt exist in C++ standard.

Actually, it would be better to specify a type to Clamp directly:
```cpp
Clamp<std::uint8_t>(0, size.value(), 25)
```

Reply:
you still have to change `ui8` to something like above to fix all build errors